src/Integration.Vsix/SonarLintTracker/IssuesSnapshot.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace SonarLintTracker
+{
+    class IssuesSnapshot : WpfTableEntriesSnapshotBase
+    {
+        private readonly string filePath;
+        private readonly int versionNumber;
+
+        // We're not using an immutable list here but we cannot modify the list in any way once we've published the snapshot.
+        public readonly List<IssueMarker> IssueMarkers = new List<IssueMarker>();
+
+        public IssuesSnapshot NextSnapshot;
+
+        internal IssuesSnapshot(string filePath, int versionNumber)
+        {
+            this.filePath = filePath;
+            this.versionNumber = versionNumber;
+        }
+
+        public override int Count
+        {
+            get
+            {
+                return this.IssueMarkers.Count;
+            }
+        }
+
+        public override int VersionNumber
+        {
+            get
+            {
+                return versionNumber;
+            }
+        }
+
+        public override bool TryGetValue(int index, string columnName, out object content)
+        {
+            if ((index >= 0) && (index < this.IssueMarkers.Count))
```

Comment:
I guess you could reduce the nesting here by inverting and returning. That's a pretty long method and I am not sure just by reading on GitHub.